pkg/model/workflow.go

```diff
@@ -79,22 +80,42 @@ type WorkflowDispatch struct {
 }
 
 func (w *Workflow) WorkflowDispatchConfig() *WorkflowDispatch {
-	if w.RawOn.Kind != yaml.MappingNode {
-		return nil
-	}
-
-	var val map[string]yaml.Node
-	if !decodeNode(w.RawOn, &val) {
-		return nil
-	}
+	switch w.RawOn.Kind {
+	case yaml.ScalarNode:
+		var val string
+		err := w.RawOn.Decode(&val)
+		if err != nil {
+			log.Fatal(err)
+			return nil
+		}
```

Comment:
Would you mind changing it to `decodeNode` here? It will be more friendly with Gitea since `log.Fatal(err)` will cause Gitea to shutdown. See also #1705

Reply:
done